### PR TITLE
[Improve][Seatunnel-Cluster] Run the server through  daemon

### PR DESCRIPTION
**Pull request by liugddx**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liugddx** [comment]
image
image

**Review by lightzhao — DISMISSED** [review]

**Comment by TyrantLucifer** [comment]
Please fix CI error

**Review by TyrantLucifer — COMMENTED** [review]
`seatunnel-cluster.sh` dose not only has the one function to start cluster node, it also has the function of showing help message like the following shown:

image

So if using this solution user will not see the help message realtime.

**Comment by TyrantLucifer** [comment]
> `seatunnel-cluster.sh` dose not only has the one function to start cluster node, it also has the function of showing help message like the following shown:
> 
> image
> 
> So if using this solution user will not see the help message realtime.

cc @Hisoka-X @EricJoy2048

**Comment by Hisoka-X** [comment]
yes, you should put `-d` to start cluster with daemon mode.

> > `seatunnel-cluster.sh` dose not only has the one function to start cluster node, it also has the function of showing help message like the following shown:
> > image
> > So if using this solution user will not see the help message realtime.
> 
> cc @Hisoka-X @EricJoy2048

**Comment by liugddx** [comment]
> yes, you should put `-d` to start cluster with daemon mode.
> 
> > > `seatunnel-cluster.sh` dose not only has the one function to start cluster node, it also has the function of showing help message like the following shown:
> > > image
> > > So if using this solution user will not see the help message realtime.
> > 
> > 
> > cc @Hisoka-X @EricJoy2048

Agree with you

**Comment by liugddx** [comment]
image

**Comment by TyrantLucifer** [comment]
> image

If user input `seatunnel.sh -h -d` or `seatunnel.sh -xxx -d`, the output log of option `-xxx` will be rewrited in files too.

**Comment by Hisoka-X** [comment]
image
image

I use your changed in local, report this error when SeaTunnel don't have `seatunnel-server.out`.

**Comment by liugddx** [comment]
> > image
> 
> If user input `seatunnel-cluster.sh -h -d` or `seatunnel-cluster.sh -xxx -d`, the output log of option `-xxx` will be rewrited in files too.

We can not use `-d` if it is an interactive argument. So `seatunnel-cluster.sh -h -d` is not allowed, but the internal parameters of the program can be parsed, like `seatunnel-cluster.sh -cn test_seatunnel -d`, Have any good Suggestions?



> image
> 
> image
> 
> I use your changed in local, report this error when SeaTunnel don't have `seatunnel-server.out`.

I'm in `deployment.md` illustrates this .
```
mkdir -p $SEATUNNEL_HOME/logs
touch $SEATUNNEL_HOME/logs/seatunnel-server.out
./bin/seatunnel-cluster.sh -d
```

File: docs/en/seatunnel-engine/deployment.md [hunk 1]
```diff
@@ -235,9 +235,12 @@ hazelcast-client:
 
 ## 7. Start SeaTunnel Engine Server Node
 
+Can be started by a daemon with `-d`.
+
 ```shell
 mkdir -p $SEATUNNEL_HOME/logs
-nohup bin/seatunnel-cluster.sh 2>&1 &
+touch $SEATUNNEL_HOME/logs/seatunnel-server.out
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/deployment.md]: Why not put it in your shell command, to reduce operation which user need do.

Comment (liugddx) [reply docs/en/seatunnel-engine/deployment.md]: Well, I think so

**Review by Hisoka-X — APPROVED** [review]
LGTM

File: seatunnel-core/seatunnel-starter/src/main/bin/seatunnel-cluster.sh [hunk 2]
```diff
@@ -69,7 +70,8 @@ do
   if [[ "${i}" == *"JvmOption"* ]]; then
     JVM_OPTION="${i}"
     JAVA_OPTS="${JAVA_OPTS} ${JVM_OPTION#*=}"
-    break
+  elif [[ "${i}" == "-d" ]]; then
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-starter/src/main/bin/seatunnel-cluster.sh]: Please make sure if user use `-h` or `--help`, the help msg must print on the console. At now if user use `./bin/seatunnel-cluster.sh -h -d`, the console will not print anything. By the way, user use `-h` can't know how to use daemon mode (please add it into `CommandArgs`)

**Review by Hisoka-X — CHANGES_REQUESTED** [review]

File: seatunnel-core/seatunnel-starter/src/main/java/org/apache/seatunnel/core/starter/seatunnel/args/ServerCommandArgs.java [hunk 3]
```diff
@@ -33,6 +33,11 @@ public class ServerCommandArgs extends CommandArgs {
             description = "The name of cluster")
     private String clusterName = "seatunnel_default_cluster";
 
+    @Parameter(
+            names = {"-d", "--daemon"},
+            description = "The cluster daemon mode")
+    private boolean daemonMod = false;
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-starter/src/main/java/org/apache/seatunnel/core/starter/seatunnel/args/ServerCommandArgs.java]:
```suggestion
    private boolean daemonMode = false;
```

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**